src/graph/SessionManager.cpp

```diff
@@ -52,20 +57,109 @@ std::shared_ptr<session::Session> SessionManager::createSession() {
     }
     DCHECK_NE(sid, 0L);
     auto session = session::Session::create(sid);
-    activeSessions_[sid] = session;
+    auto now = time::WallClock::fastNowInSec();
+    activeSessions_[sid] = std::make_pair(session, now);
+    activeSessForMeta_[sid] = now;
     session->charge();
+
+    // sessionId, FLAGS_graph_server_ip:FLAGS_port, start_time
+    executor_->add([sid, now, this] () {
+        this->doAddSession(sid, now);
+    });
+
     return session;
```

Comment:
If the metad is abnormal, here also return the session?

Reply:
Yes.
Graphd manages the local session, even if metad is not running, you can connect to graphd through the console  and create a session

Reply:
So here use  a background thread  to do `doAddSession `.
 Of course there must be a time delay.